lib/_stream_writable.js

```diff
@@ -194,23 +194,16 @@ function writeAfterEnd(stream, cb) {
   process.nextTick(cb, er);
 }
 
-// If we get something that is not a buffer, string, null, or undefined,
-// and we're not in objectMode, then that's an error.
-// Otherwise stream chunks are all considered to be of length=1, and the
-// watermarks determine how many objects to keep in the buffer, rather than
-// how many bytes or characters.
 function validChunk(stream, state, chunk, cb) {
   var valid = true;
   var er = false;
-  // Always throw error if a null is written
-  // if we are not in object mode then throw
-  // if it is not a buffer, string, or undefined.
+  // Always throw error if a null is written.
+  // If we are not in object mode then throw if it is not a string or undefined.
   if (chunk === null) {
     er = new TypeError('May not write null values to stream');
-  } else if (!(chunk instanceof Buffer) &&
-      typeof chunk !== 'string' &&
-      chunk !== undefined &&
-      !state.objectMode) {
+  } else if (typeof chunk !== 'string' &&
```

Comment:
Would switching to `Buffer.isBuffer()` have the same performance profile as `instanceof`? By removing the `instanceof Buffer` check here, this becomes a semver-major. If we can switch to `Buffer.isBuffer()` and still get a performance boost, then this can be semver-patch.